tests/include/test/macros.h

```diff
@@ -123,36 +123,36 @@
  *                  This expression may be evaluated multiple times.
  *
  */
-#define ASSERT_ALLOC(pointer, length)                           \
-    do                                                            \
-    {                                                             \
-        TEST_ASSERT((pointer) == NULL);                       \
-        if ((length) != 0)                                     \
-        {                                                         \
-            (pointer) = mbedtls_calloc(sizeof(*(pointer)), \
+#define TEST_CALLOC_OR_FAIL(pointer, length)                \
```

Comment:
We should call the second parameter `item_count` or some such rather than `length`. At least for people who use IDEs with fancy popups, it would convey the fact that it's a length in items and not bytes.